src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -68,21 +71,31 @@
                                       ((:on-rename props) name)
                                       (js/alert (str "Exception:\n" (.-statusText xhr)))))}))))})))
 
-
 (react/defc MethodDetailsViewer
-  {:render
-   (fn [{:keys [state]}]
+  {:get-fields
+   (fn [{:keys [refs]}]
+     (react/call :get-fields (@refs "methodDetails")))
+   :render
+   (fn [{:keys [props state]}]
      (cond
-       (:loaded-method @state) [comps/EntityDetails {:entity (:loaded-method @state)}]
+       (:loaded-method @state)
+       [comps/EntityDetails {:entity (:loaded-method @state)
+                             :editing? (:editing? props)
+                             :ref "methodDetails"
+                             :onSnapshotIdChange (:onSnapshotIdChange props)
+                             :snapshots ((:methods props) (map (:loaded-method @state) ["namespace" "name"]))}]
        (:error @state) (style/create-server-error-message (:error @state))
-       :else [comps/Spinner {:text "Loading details..."}]))
+        :else [comps/Spinner {:text "Loading details..."}]))
    :component-did-mount
-   (fn [{:keys [props state]}]
+   (fn [{:keys [this props state]}]
+     (react/call :load-agora-method this props state))
+   :load-agora-method
+   (fn [{:keys [props state]} method-ref]
      (endpoints/call-ajax-orch
        {:endpoint (endpoints/get-agora-method
```

Comment:
Note to self: this function should probably just take a map—I'm not sure there is value in extracting these keys into positional parameters.